javacord-core/src/main/java/org/javacord/core/entity/channel/RegularServerChannelUpdaterDelegateImpl.java

```
@@ -0,0 +1,147 @@
+        implements RegularServerChannelUpdaterDelegate {
+
+    /**
+     * The channel to update.
+     */
+    protected final RegularServerChannel channel;
+
+    /**
+     * The position to update.
+     */
+    protected Integer position = null;
+
+    /**
+     * A map with all overwritten user permissions.
+     */
+    protected Map<Long, Permissions> overwrittenUserPermissions = null;
+
+    /**
+     * A map with all overwritten role permissions.
+     */
+    protected Map<Long, Permissions> overwrittenRolePermissions = null;
+
+    /**
+     * Creates a new server channel updater delegate.
+     *
+     * @param channel The channel to update.
+     */
+    public RegularServerChannelUpdaterDelegateImpl(RegularServerChannel channel) {
+        super(channel);
+        this.channel = channel;
+    }
+
+
+    @Override
+    public void setRawPosition(int rawPosition) {
+        this.position = rawPosition;
+    }
+
+    @Override
+    public <T extends Permissionable & DiscordEntity> void addPermissionOverwrite(T permissionable,
+                                                                                  Permissions permissions) {
+        populatePermissionOverwrites();
+        if (permissionable instanceof Role) {
+            overwrittenRolePermissions.put(permissionable.getId(), permissions);
+        } else if (permissionable instanceof User) {
+            overwrittenUserPermissions.put(permissionable.getId(), permissions);
+        }
+    }
+
+    @Override
+    public <T extends Permissionable & DiscordEntity> void removePermissionOverwrite(T permissionable) {
+        populatePermissionOverwrites();
+        if (permissionable instanceof Role) {
+            overwrittenRolePermissions.remove(permissionable.getId());
+        } else if (permissionable instanceof User) {
+            overwrittenUserPermissions.remove(permissionable.getId());
+        }
+    }
+
+    @Override
+    public CompletableFuture<Void> update() {
```

Why is this overwritten?
The only difference to the superclass is, that it does not support audit log reason.
Does it not support audit log reason?
Then it would be bad that you can still set it on the updater but it is just silently ignored.